### PR TITLE
Feature/rnp 96

### PR DESCRIPTION
**Pull request by MichielVrins**

### CODE REVIEW

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/module/RNOneginiSdkModule.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@Module
+class RNOneginiSdkModule(private val reactApplicationContext: ReactApplicationContext) {
+
+  @Provides
+  @Singleton
+  fun provideContext() = reactApplicationContext.applicationContext
+
+  @Provides
+  @Singleton
+  fun provideReactApplicationContext() = reactApplicationContext
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/module/RNOneginiSdkModule.kt:18]: I think we don't need reactNativeApplication context to be injected anywhere

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/OneginiSDK.kt [hunk 2]
```diff
@@ -13,24 +13,22 @@ import com.onegini.mobile.sdk.reactnative.handlers.mobileauthotp.MobileAuthOtpRe
...
-    lateinit var pinAuthenticationRequestHandler: PinAuthenticationRequestHandler
-        private set
-    lateinit var createPinRequestHandler: CreatePinRequestHandler
-        private set
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class OneginiSDK @Inject constructor(
+    private val reactApplicationContext: ReactApplicationContext,
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/OneginiSDK.kt:21]: application context can be injected directly here:
`private val applicationContext: Context`

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/network/AnonymousService.kt [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
+import javax.inject.Inject
+import javax.inject.Singleton
 
-class AnonymousService private constructor() {
-
-    private val applicationDetailsRetrofitClient: ResourcesClient = OneginiComponents.secureResourceClient.prepareSecuredAnonymousRetrofitClient(ResourcesClient::class.java)
-
+@Singleton
+class AnonymousService @Inject constructor(private val secureResourceClient: SecureResourceClient){
+    private val applicationDetailsRetrofitClient: ResourcesClient by lazy {
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/network/AnonymousService.kt:15]: correct retrofit client can also be injected. You can create new dagger module that will have provider methods and each can provide different retrofit client. Each provider would return the same type, thus you will need to use @Named annotation, so dagger knows which one to inject

**Review by aleknow — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/network/UserService.kt [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
 import javax.inject.Inject
+import javax.inject.Named
 import javax.inject.Singleton
 
 @Singleton
-class UserService @Inject constructor(private val secureResourceClient: SecureResourceClient){
-    private val applicationDetailsRetrofitClient: ResourcesClient by lazy {
-        secureResourceClient.prepareSecuredUserRetrofitClient(ResourcesClient::class.java)
-    }
+class UserService  @Inject constructor(@Named(Constants.USER_SERVICE) private val userRetrofitClient: Lazy<ResourcesClient>){
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/network/UserService.kt:16]: one too many space `UserService  @Inject`

Comment (aleknow) [reply android/src/main/java/com/onegini/mobile/sdk/reactnative/network/UserService.kt:16]: You can add import for Constants so there only `@Named(USER_SERVICE)`

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/network/ImplicitUserService.kt [hunk 5]
```diff
@@ -1,26 +1,27 @@
...
 import javax.inject.Inject
+import javax.inject.Named
 import javax.inject.Singleton
 
 @Singleton
-class ImplicitUserService @Inject constructor(private val secureResourceClient: SecureResourceClient){
-    private val applicationDetailsRetrofitClient: ResourcesClient by lazy {
-        secureResourceClient.prepareSecuredImplicitUserRetrofitClient(ResourcesClient::class.java)
-    }
+class ImplicitUserService @Inject constructor(@Named(Constants.IMPLICIT_USER_SERVICE) private val implicitRetrofitClient: Lazy<ResourcesClient>){
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/network/ImplicitUserService.kt:16]: You can add import for Constants so there only `@Named(IMPLICIT_USER_SERVICE)`

File: android/src/main/java/com/onegini/mobile/sdk/reactnative/network/AnonymousService.kt [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
 import javax.inject.Inject
+import javax.inject.Named
 import javax.inject.Singleton
 
 @Singleton
-class AnonymousService @Inject constructor(private val secureResourceClient: SecureResourceClient){
-    private val applicationDetailsRetrofitClient: ResourcesClient by lazy {
-        secureResourceClient.prepareSecuredAnonymousRetrofitClient(ResourcesClient::class.java)
-    }
+class AnonymousService @Inject constructor(@Named(Constants.ANONYMOUS_SERVICE) private val anonymousRetrofitClient: Lazy<ResourcesClient>) {
```
Comment (aleknow) [inline android/src/main/java/com/onegini/mobile/sdk/reactnative/network/AnonymousService.kt:16]: You can add import for Constants so there only `@Named(ANONYMOUS_SERVICE)`

**Review by aleknow — CHANGES_REQUESTED** [review]

File: android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt [hunk 7]
```diff
@@ -1,178 +0,0 @@
...
-import io.mockk.mockkStatic
-import org.junit.Before
-import org.junit.Test
-import org.junit.runner.RunWith
-import org.mockito.Mock
-import org.mockito.junit.MockitoJUnitRunner
-import org.mockito.kotlin.verify
-
-@RunWith(MockitoJUnitRunner::class)
-class OneginiSdkWrapperTests {
```
[the remark this answers is not in the document]
Comment (MichielVrins) [reply android/src/test/java/com/onegini/mobile/sdk/reactnative/OneginiSdkWrapperTests.kt:32]: We will be removing the SdkWrapper.kt anyway when we move everything into useCases as we can link them up in the RNOneginiSdk class. For now these tests don't actually add any value and are just a nuisance when moving methods into useCases.

**Review by aleknow — APPROVED** [review]

**Pull request merged**